solara/components/datatable.py

```diff
@@ -100,12 +100,12 @@ def DataTable(
     i2 = min(total_length, (page + 1) * items_per_page)
 
     columns = use_df_column_names(df)
-
+    rows = use_df_row_names(df)
     items = []
     dfs = df_slice(df, i1, i2)
     records = df_records(dfs)
     for i in range(i2 - i1):
-        item = {"__row__": i + i1}  # special key for the row number
+        item = {"__row__": rows[i + i1]}  # special key for the row number
```

Comment:
If a pandas dataframe index can contain anything, we maybe should call format on it as well,

Reply:
I don't understand. You are saying we should call format() on rows[i+1]? As in; item = {"__row__": format(rows[i + i1])} ?